solver/src/pending_transactions.rs

```diff
@@ -32,18 +39,35 @@ pub enum Fee {
     },
 }
 
-// Get the pending transactions from a node.
-//
-// This works better than rust-web3's `block_with_txs` because some nodes set the `miner` field to
-// `null`. This does not follow the ethrpc specification and fails deserialization in rust-web3.
+// Get pending transactions from a mempool.
 pub async fn pending_transactions(transport: &DynTransport) -> Result<Vec<Transaction>> {
-    let params = vec!["pending".into(), true.into()];
-    let response = transport
-        .execute("eth_getBlockByNumber", params)
+    match transport
+        .execute("txpool_content", Default::default())
         .await
-        .context("transport failed")?;
-    let block: Block = serde_json::from_value(response).context("deserialize failed")?;
-    Ok(block.transactions)
+        .context("transport failed")
+    {
+        Ok(response) => {
+            let txpool: TxPool = serde_json::from_value(response).context("deserialize failed")?;
+
+            Ok(txpool
+                .pending
```

Comment:
Seeing as this is used for getting any submitted transaction from a previously executing solver, maybe it makes sense to return both `pending` and `queued` transactions? I was trying to understand exactly what the difference between these two was but it wasn't super obvious to me from the [documentation](https://geth.ethereum.org/docs/rpc/ns-txpool#txpool_content).

Reply:
The difference between pending and queued transaction from mempool is that pending transaction are ready to be included in a block, while queued transactions cannot be included in a block at the time, because for example, there might be a nonce gap between the queued transaction and last transaction sent from the same address, so they are queued for future inclusion.
Added merging of pending and queued transactions as suggested, to be sure that our transaction is not missed by any chance.